packages/compass-saved-aggregations-queries/src/components/open-item-modal.tsx

```diff
@@ -161,6 +168,13 @@ const OpenItemModal: React.FunctionComponent<OpenItemModalProps> = ({
             label="Collection"
           ></CollectionSelect>
         </div>
+        <Checkbox
+          className={checkbox}
+          checked={updateNamespace}
+          onChange={() => setUpdateNamespace(!updateNamespace)}
```

Comment:
Small nit: usually when you are dealing with state updates based on UI in React, you would generally want to sync it directly from the input element, not from the state in the current render closure to avoid state going out of sync between what's being rendered by the browser and what is getting passed to the state update. The difference is subtle and it doesn't chance much here specifically, but it's a rule of thumb that I'd suggest to just keep in mind.

In this particular case it would mean changing the `onChange` handler to use the event argument passed to the callback to get the value

```jsx
onChange={(event) => {
  setUpdateNamespace(event.target.checked);
}}
```

Also just generally speaking, not related to handling DOM events, when your `setState` depends on the previous state, the pattern is to use the setState callback instead, this guarantees that you are working with the previous state when doing an update and not a potentially stale value from the render closure:

```
setUpdateNamespace(prevState => !prevState)
```